Purge Bucket - Fixes & Features Additions In Macros

edwardyeeks opened the pull request:
<!--
Thank you for your interest in contributing to the VoronUsers repository, it is
highly appreciated!

**Please make sure the submission conforms to the rules outlined below. PRs which fail
to conform to the rules below are likely to be rejected.**
-->

  * [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folders names MUST NOT contain spaces. If possible, make sure file names also 
    do not contain any spaces.
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.

<!--
Describe the submission further using the template provided below. The more 
details the better!
-->

#### Which mods/configurations/profiles are added by this PR?
Decontaminator Purge Bucket & Nozzle Scrubber. Fix error in script defining random selection of bucket halves for purging. The issue was that when selecting the right half of the bucket to randomly purge, it would put the nozzle too close/within the brush; it should now be fixed. Small speed unit correction. (thanks Deutherius#3295 on VD discord for pointing these out).

jurisv commented on printer_mods/edwardyeeks/Decontaminator_Purge_Bucket_&_Nozzle_Scrubber/Macros/nozzle_scrub.cfg:
```diff
@@ -161,7 +161,7 @@ gcode:
 
          ### Position for purge. Randomly selects middle of left or right bucket. It references from the middle of the left bucket.
-         G1 X{bucket_start + (bucket_left_width / 2) + (bucket_pos * bucket_gap) + (bucket_pos * (bucket_right_width / 2))}
+         G1 X{bucket_start + (bucket_left_width / (2 - bucket_pos) + (bucket_pos * bucket_gap) + (bucket_pos * (bucket_right_width / 2))}
```
The brackets are not balanced causing syntax error

jurisv replied:
I think it should be:
`G1 X{bucket_start + (bucket_left_width / (2 - bucket_pos)) + (bucket_pos * bucket_gap) + (bucket_pos * (bucket_right_width / 2))}`

edwardyeeks replied:
You're right, thanks for picking it up!

edwardyeeks commented:
Another mistake or two found along the way that have been fixed! Also, a few small features have been added in the macro, namely:

- Scrubbing location and direction now follow the randomized location of left and right bucket. This removes any unnecessary travel that previously existed when transitioning from purging to scrubbing. Also adds a bit more character to the macro!
- Added filament retract after a purge. Stops filament from continuously oozing due to back pressure even after a scrub is performed.
- Added a dwell/wait time after a purge for particularly oozy filaments.

Edward
